serve/mlc_serve/model/tvm_model.py

```diff
@@ -330,7 +331,8 @@ def generate(
             cache.pending_copy_from_to = []
 
         try:
-            next_tokens = sample(logits, sampling_params, self.vocab_size)
+            next_tokens, logprob_infos = sample(logits, sampling_params, self.vocab_size)
+            current_ids = list(input_ids.numpy())
```

Comment:
This seems incorrect. `input_ids` is a concatenation of a batch of token ids. So attaching this list as `previous_tokens` doesn't make sense.

Moreover, you shouldn't be doing numpy conversion and list creation naively like this. As I said repeatedly, there should be no perf regression when logprob is not used.